### PR TITLE
Added source pkg to pkg build pipelines

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

Added source packages (tar.gz) to the package build pipelines. Removed `SPECS` folder in favor of GitHub Release drops.

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: devops/pipeline/binary_package.yaml [hunk 1]
```diff
@@ -15,191 +15,232 @@ trigger:
...
+
+    - script: |
+        docker exec $(CONTAINER_ID) bash -c "gcovr -r /AzOsConfig/src . -e /AzOsConfig/src/agents/pnp/ --xml /AzOsConfig/build-$(arch)/coverage.xml"
+      displayName: Run code coverage
+      condition: eq(variables.coverage, true)
+      workingDirectory: $(Agent.BuildDirectory)
+
+    # Needed for the code coverage report task below (See `PublishCodeCoverageResults`)
+    - task: UseDotNet@2
+      displayName: Use .NET Core sdk
```
Comment (MariusNi) [inline devops/pipeline/binary_package.yaml:231]: >sdk [](http://example.com/codeflow?start=33&length=3)

nit: "sdk" shouldn't be "SDK"?

File: devops/pipeline/binary_package.yaml [hunk 2]
```diff
@@ -15,191 +15,232 @@ trigger:
...
+          ignoreTestFailures: false
+
+    pool:
+      vmImage: 'ubuntu-20.04'
+
+    steps:
+
+    - checkout: self
+      submodules: recursive
+      clean: true
```
Comment (MariusNi) [inline devops/pipeline/binary_package.yaml:139]: From here to end of file - are these just formatting changes? And if so, why are they necessary? Blocks appear the same with exception of being spaced to the right (why?)

Comment (AhmedBM) [reply devops/pipeline/binary_package.yaml:139]: yes, the new stuff is from 19-52 i added a new `source` stage which produces and publishes the source package to the pipeline

Comment (MariusNi) [reply devops/pipeline/binary_package.yaml:139]: That part is fine. Question is about the changes from here on line 136 to end of file - are these really necessary?

**Review by MariusNi — CHANGES_REQUESTED** [review]
:clock1:

**Review by MariusNi — DISMISSED** [review]
:shipit:

File: devops/pipeline/binary_package.yaml (hunk 2 above)
Comment (AhmedBM) [reply devops/pipeline/binary_package.yaml:139]: yes, they need to be indented as we've introduced a more steps though

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**